nncf/experimental/post_training/graph/model_transformer.py

```diff
@@ -0,0 +1,129 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from abc import ABC
+from abc import abstractmethod
+from typing import List
+from typing import TypeVar
+
+from nncf.common.graph.model_transformer import ModelTransformer
+from nncf.common.graph.transformations.commands import TransformationCommand
+from nncf.common.graph.transformations.layout import TransformationLayout
+from nncf.experimental.post_training.statistics.statistic_point import StatisticPointsContainer
+
+ModelType = TypeVar('ModelType')
+
+
+# pylint: disable=no-member
+class StaticModelTransformerBase(ModelTransformer, ABC):
+    QUANTIZER_NAME_PREFIX = 'QuantizeLinear_'
+    DEQUANTIZER_NAME_PREFIX = 'DequantizeLinear_'
+    SCALE_TENSOR_NAME_PREFIX = 'scale_'
+    ZERO_POINT_NAME_PREFIX = 'zero_point_'
+
+    def __init__(self, model: ModelType):
+        super().__init__(model)
+        self._transformation_layout = None
+        self._transformations_list = []
+
+        # Transformation commands list
+        self._output_insertion_command = None
+        self._quantizer_insertion_command = None
+
+        self._callbacks_by_commands = {
```

Comment:
Not good as well. Methods should be statically defined to allow good code navigation and typing. This also forces you to use `type` further down the line, something which you won't see anywhere in the arch design of the NNCF major classes.

I suggest to define the `_transform(...)` virtual method in this abstract class and make the public `transform(...)` method call that instead. You then define the `_transform(...)` in the concrete subclasses with the correct backend-specific functionality.